File: js/100doors/doorWalker.js
var Door = require('./door.js');

function DoorWalker() {}

DoorWalker.prototype.walk = function (doors, step) {

    if (step <= 0) {
        return;
    }

    step = step || 1;

    if (doors === undefined) {
        return;
    }

    if (doors.length === 0) {
        return;
    }

    for (var i = step - 1; i < doors.length; i += step) {
        doors[i].toggle();
    }
}

module.exports = DoorWalker;

File: python/100doors/doorwalker.py
#!/usr/bin/python3

class DoorWalker:
    """ DoorWalker class """

    def __init__(self):
        """ (DoorWlaker) -> NoneType """

    def walk(self, doors):
        """ (DoorWalker, list of Door) -> NoneType
        Walk throught the doors and change the door state.
        """

        for door in doors:
            door.toggle()

if (__name__ == '__main__'):
    walker = DoorWalker()
    print(walker) 
